Update docs to use ARM based images for Apple M1/M2 Macbooks (#45142)
ray-project/ray: doc/source/cluster/kubernetes/troubleshooting/troubleshooting.md

# Contents

- [Use ARM-based docker images for Apple M1/M2 Macbooks](#docker-image-for-apple-macbooks) 
- [Upgrade KubeRay](#upgrade-kuberay)
- [Worker init container](#worker-init-container)
- [Cluster domain](#cluster-domain)
- [RayService](#rayservice)
- [Autoscaler](#autoscaler)
- [Other questions](#other-questions)

(docker-image-for-apple-macbooks)=

> ## Use ARM-based docker images for Apple M1/M2 Macbooks

Reviewer: not sure we even want to support this. we never really tested kuberay on arm..

Reviewer 2: We haven't tested KubeRay on ARM in CI, but KubeRay builds a multi-architecture image that supports both `linux/amd64` and `linux/arm64`. We conducted some CUJ analyses for the KubeRay documentation and found that almost all users struggle to run an easy KubeRay example on Mac M1 because they need to update to an ARM64-compatible Ray image. That's why I ask @shaikhi to open this PR. Thank @aslonnie for the review!

Reviewer: does kuberay prefer to have multi-arch ray images? we can build those, it is not actually not very hard.

the more tricky / expensive part is testing. like what kind of quality / reliability bar we want to reach for this?

Reviewer 2: > does kuberay prefer to have multi-arch ray images? 

It would be very helpful for UX! The multi-architecture image can increase the success rate for users following the KubeRay documentation. We only need to ensure that users can easily follow the documents with the multi-architecture image, and we still recommend that users use the original images for production.

Reviewer 3: if we decide to build them and **not** test them we need to be explicit in documentation on docs.ray.io, and github. who can take that action item to update?

Author: I think only this comment/note is unresolved from @anyscalesam :
> if we decide to build them and not test them we need to be explicit in documentation on docs.ray.io, and github. who can take that action item to update?

I'm not sure what to do for this? Can we create another ticket for this?

Reviewer 2: @shaikhismail I will take that part after the multi-platform images are out. I think the PR is ready to merge.

Currently Ray builds different images for different platforms. Until Ray moves to building multi-architecture images, [tracked by this Github issue](https://github.com/ray-project/ray/issues/39364), the solution here is to use platform specific docker images in the head and worker group specs of the [RayCluster config](https://docs.ray.io/en/latest/cluster/kubernetes/user-guides/config.html#image). 

> For e.g. just using `image: rayproject/ray:2.10.0` led to issues such as head service (`rayservice-sample-head-svc`) and Ray Serve service (`rayservice-sample-serve-svc`) for the RayService custom resource not launching in the head node and Serve applications not getting ready. The solution is to use `image: rayproject/ray:2.10.0-aarch64` instead for Apple M1/M2 Macbooks. [Link to issue details and discussion](https://ray-distributed.slack.com/archives/C02GFQ82JPM/p1712267296145549).

Reviewer 2: I think we don't need to mention the details 

> issues such as head service (`rayservice-sample-head-svc`) and Ray Serve service (`rayservice-sample-serve-svc`) for the RayService custom resource not launching in the head node and Serve applications not getting ready.

We can just mention that you need to use an image with the tag "aarch64" (e.g. `rayproject/ray:2.10.0-aarch64`) if you are running on a Mac M1 or M2.

Author: addressed this with another commit.

# Contents

> - [Use ARM-based docker images for Apple M1/M2 Macbooks](#docker-image-for-apple-macbooks) 

Reviewer 4: ```suggestion
- [Use ARM-based docker images for Apple M1 or M2 MacBooks](#docker-image-for-apple-macbooks) 
```

- [Upgrade KubeRay](#upgrade-kuberay)
- [Worker init container](#worker-init-container)
- [Cluster domain](#cluster-domain)
- [RayService](#rayservice)
- [Autoscaler](#autoscaler)
- [Other questions](#other-questions)

(docker-image-for-apple-macbooks)=

> ## Use ARM-based docker images for Apple M1/M2 Macbooks

Reviewer 4: ```suggestion
## Use ARM-based docker images for Apple M1 or M2 MacBooks
```

> Currently Ray builds different images for different platforms. Until Ray moves to building multi-architecture images, [tracked by this Github issue](https://github.com/ray-project/ray/issues/39364), the solution here is to use platform specific docker images in the head and worker group specs of the [RayCluster config](https://docs.ray.io/en/latest/cluster/kubernetes/user-guides/config.html#image). 

Reviewer 4: ```suggestion
Ray builds different images for different platforms. Until Ray moves to building multi-architecture images, [tracked by this Github issue](https://github.com/ray-project/ray/issues/39364), use platform-specific docker images in the head and worker group specs of the [RayCluster config](https://docs.ray.io/en/latest/cluster/kubernetes/user-guides/config.html#image). 
```

> Use an image with the tag `aarch64` (e.g. `image: rayproject/ray:2.12.0-aarch64`) if you are running KubeRay on a Mac M1 or M2.

Reviewer: can we change this to `2.20.0`?

also, this is assuming user has some form of docker installed on the mac (which is often based on a Linux virtual machine)? Should you mention how to install docker?

Reviewer 2: > Should you mention how to install docker?

I think we may not need to mention it at this moment.

Author: updated to `2.20.0`, thanks for the review.

> Use an image with the tag `aarch64` (e.g. `image: rayproject/ray:2.12.0-aarch64`) if you are running KubeRay on a Mac M1 or M2.

Reviewer 4: ```suggestion
Use an image with the tag `aarch64`, for example, `image: rayproject/ray:2.12.0-aarch64`), if you are running KubeRay on a MacBook M1 or M2.
```

[Link to issue details and discussion](https://ray-distributed.slack.com/archives/C02GFQ82JPM/p1712267296145549).

(upgrade-kuberay)=
## Upgrade KubeRay

> If you have issues upgrading KubeRay, refer to the [upgrade guide](#kuberay-upgrade-guide).

Reviewer 4: ```suggestion
If you have issues upgrading KubeRay, see the [upgrade guide](#kuberay-upgrade-guide).
```
